pywr/parameters/_thresholds.pyx

```diff
@@ -0,0 +1,203 @@
+    def load(cls, model, data):
+        node = model._get_node_from_ref(model, data.pop("storage_node"))
+        threshold = data.pop("threshold")
+        values = data.pop("values", None)
+        predicate = data.pop("predicate", None)
+        return cls(model, node, threshold, values=values, predicate=predicate, **data)
+StorageThresholdParameter.register()
+
+
+cdef class NodeThresholdParameter(AbstractThresholdParameter):
+    """ Returns one of two values depending on current flow in a node
+
+    Parameters
+    ----------
+    recorder : `pywr.core.AbstractNode`
+
+    """
+    def __init__(self, model, AbstractNode node, *args, **kwargs):
+        super(NodeThresholdParameter, self).__init__(model, *args, **kwargs)
+        self.node = node
+
+    cpdef double _value_to_compare(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+        return self.node._flow[scenario_index._global_id]
+
+    @classmethod
+    def load(cls, model, data):
+        node = model._get_node_from_ref(model, data.pop("node"))
+        threshold = data.pop("threshold")
+        values = data.pop("values", None)
+        predicate = data.pop("predicate", None)
+        return cls(model, node, threshold, values=values, predicate=predicate, **data)
+NodeThresholdParameter.register()
+
+
+cdef class ParameterThresholdParameter(AbstractThresholdParameter):
+    """ Returns one of two values depending on the value of a Parameter
+
+    Parameters
+    ----------
+    recorder : `pywr.core.AbstractNode`
+
+    """
+    def __init__(self, model, Parameter param, *args, **kwargs):
+        super(ParameterThresholdParameter, self).__init__(model, *args, **kwargs)
+        self.param = param
+        self.children.add(param)
+
+    cpdef double _value_to_compare(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+        return self.param.get_value(scenario_index)
+
+    @classmethod
+    def load(cls, model, data):
+        param = load_parameter(model, data.pop('parameter'))
+        threshold = data.pop("threshold")
+        values = data.pop("values", None)
+        predicate = data.pop("predicate", None)
+        return cls(model, param, threshold, values=values, predicate=predicate, **data)
+ParameterThresholdParameter.register()
+
+
+cdef class RecorderThresholdParameter(AbstractThresholdParameter):
+    """Returns one of two values depending on a Recorder value and a threshold
+
+    Parameters
+    ----------
+    recorder : `pywr.recorder.Recorder`
+
+    """
+
+    def __init__(self,  model, Recorder recorder, *args, **kwargs):
+        super(RecorderThresholdParameter, self).__init__(model, *args, **kwargs)
+        self.recorder = recorder
+        self.recorder.parents.add(self)
+
+    cpdef double _value_to_compare(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+        # TODO Make this a more general API on Recorder
+        return self.recorder.data[timestep._index - 1, scenario_index._global_id]
```

Comment:
I'm currently using this recorder with "MeanFlowRecorder". I would need to implement `.values` on that, but it shouldn't be difficult.